File: app/src/main/java/com/fdi/pad/pethouse/activity_registration_name.java
package com.fdi.pad.pethouse;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class activity_registration_name extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(com.fdi.pad.pethouse.R.layout.activity_registration_name);


        Button button = findViewById(com.fdi.pad.pethouse.R.id.btnSigNom);

        button.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {

                //guaradamos los datos
            EditText nom = (EditText) findViewById(com.fdi.pad.pethouse.R.id.textPersonName);
            EditText ape = (EditText) findViewById(com.fdi.pad.pethouse.R.id.textPersonApe);

            nom.toString().trim();
            ape.toString().trim();
            if(nom.equals("") || ape.equals("")){

                Toast mensaje =
                        Toast.makeText(getApplicationContext(),
                                "Faltan datos", Toast.LENGTH_LONG);

                mensaje.show();

            }else{

                Intent intent = new Intent(activity_registration_name.this, regEmail.class);
                intent.putExtra("nombre",nom.getText());
                intent.putExtra("apellidos",ape.getText());
                startActivity(intent);

            }
            }
        });

    }
}

File: app/src/main/java/com/fdi/pad/pethouse/userRegistration/ActivityRegistrationName.kt
package com.fdi.pad.pethouse.userRegistration

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import com.fdi.pad.pethouse.R
import kotlinx.android.synthetic.main.activity_registration_name.*

/**
 * Actividad que define el paso del registro donde se introduce el nombre del usuario.
 */
class ActivityRegistrationName : AppCompatActivity() {

    /*------------------------------CONSTANTES---------------------------*/
    /**
     * Parámetro para determinar el nombre del usuario.
     */
    private val nameUser = "name"
    /**
     * Parámetro para determinar los apellidos del usuario.
     */
    private val surnameUser  = "surname"

    /*--------------------------ETAPAS---------------------------------*/
    /**
     * Creación de la actividad.
     *
     * @param savedInstanceState El estado de la aplicación en un paquete.
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registration_name)

        buttonNext.setOnClickListener { buttonNext() }
    }

    /*--------------------------MÉTODOS PRIVADOS---------------------------------*/
    /**
     * Valida los campos y envía a la siguiente pantalla.
     */
    private fun buttonNext() {
        /*Cogemos los valores de los casilleros.*/
        val name = editTextName.text.toString()
        val surname = editTextSurname.text.toString()
        if (!validateForm(name, surname)) return

        val intent = Intent(this@ActivityRegistrationName, ActivityRegistrationEmail::class.java)
        intent.putExtra(nameUser, name)
        intent.putExtra(surnameUser, surname)
        startActivity(intent)
    }

    /**
     * Comprueba que los campos de nombre y apellidos no estén vacíos.
     *
     * @return Nombre y apellidos correctos
     */
    private fun validateForm(name: String, surname: String): Boolean {
        var correctName = true
        var correctSurname = true

        if (TextUtils.isEmpty(name)) {
            editTextName.error = getString(R.string.required_field)
            correctName = false
        } else editTextName.error = null

        if (TextUtils.isEmpty(surname)) {
            editTextSurname.error = getString(R.string.required_field)
            correctSurname = false
        } else editTextSurname.error = null

        return correctName && correctSurname
    }
}